File: 2433-find-the-original-array-of-prefix-xor/2433-find-the-original-array-of-prefix-xor.java
class Solution {
    public int[] findArray(int[] pref) {
        //int ans[] =new int[pref.length];
        int prev = pref[0];
        for(int i=1; i<pref.length; i++){
            int temp = pref[i];
            pref[i] = pref[i]^prev;
            prev = temp;
            
        }
        return pref;
    }
}

File: 2433-find-the-original-array-of-prefix-xor/2433-find-the-original-array-of-prefix-xor.cs
public class Solution {
    public int[] FindArray(int[] pref) {
        int[] ans =new int[pref.Length];
        ans[0]=pref[0];
        for(int i=1; i<pref.Length; i++){
            ans[i] = pref[i]^pref[i-1];
        }
        return ans;
    }
}